akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -25,5 +25,12 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
-  val addressUid: Int = ThreadLocalRandom.current.nextInt()
-}
\ No newline at end of file
+  val longAddressUid: Long = {
+    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
```

Comment:
We might keep them as-is, we can give those only the low 32 bits. But it is better to use all 64 bits of course if possible.